processor/groupbyattrsprocessor/attribute_groups.go

```diff
@@ -187,3 +225,21 @@ func (lgba *logsGroupedByAttrs) attributeGroup(baseResource pdata.Resource, reco
 
 	return res
 }
+
+// attributeGroup searches for a group with matching attributes and returns it. If nothing is found, it is being created
+func (mgba *metricsGroupedByAttrs) attributeGroup(baseResource pdata.Resource, recordAttrs pdata.AttributeMap) pdata.ResourceMetrics {
+	res, found := mgba.findGroup(baseResource, recordAttrs)
+	if !found {
+		res = mgba.AppendEmpty()
+		baseResource.CopyTo(res.Resource())
+
+		// This prioritizes metric attributes over resource attributes, if they overlap
+		attrs := res.Resource().Attributes()
+		recordAttrs.Range(func(k string, v pdata.AttributeValue) bool {
+			attrs.Upsert(k, v)
+			return true
+		})
+	}
+
```

Comment:
Could we invert the condition here to reduce the amount of branching?

```suggestion
if found {
   return res
}
res = mgba.AppendEmpty()
baseResource.CopyTo(res.Resource())

// This prioritizes metric attributes over resource attributes, if they overlap
attrs := res.Resource().Attributes()
recordAttrs.Range(func(k string, v pdata.AttributeValue) bool {
    attrs.Upsert(k, v)
    return true
})
return res
```

Reply:
Okay, and I'll repeat the same pattern for Logs and Traces as well.